SeaLevel, Vehicle Tags, boxExtent scaling & more

sh4rkman opened the pull request:
## 1. Added "Scaling Logging" function into `Child Logger` for boxes.
Objectives capzones now have scaling_x/y/z properties, which fix #121

<img width="210" height="110" alt="image" src="https://github.com/user-attachments/assets/e6ceef80-712a-4876-ae48-965af1ea79c1" />


Example with Police Station on Anvil_RAAS_V1 : 

```
    "name": "Police Station",
    "objectDisplayName": "B5-Police Station",
    "objectName": "B5-PoliceStation",
    "objects": [
        {
            "boxExtent": {
                "extent_x": 32,
                "extent_y": 32,
                "extent_z": 32,
                "rotation_x": 0,
                "rotation_y": 0,
                "rotation_z": -154.999984741211,
                "scaling_x": 38, // new
                "scaling_y": 15, // new
                "scaling_z": 50 // new
            },
            ...
        },
```  
        
## 2. Added Vehicle Tags function to `Faction Unit Logging`

Now extract for each vehicles its tags.

Tags can tell if a vehicle can go over water, if it has an ATGM and most importantly what's his class is (Light/Medium/Heavy).
With its class + vehicle type it's possible to find its ticket value in `KillDeathRulesetEnmyTable`, but i couldn't do it for now.

<img width="210" height="120" alt="image" src="https://github.com/user-attachments/assets/d790751a-811e-486c-aaf9-f339b461865f" />


Exemple : 

```
    {
        "count": 2,
        "delay": 15,
        "icon": "map_ifv",
        "rawType": "BP_ASLAV_C",
        "respawnTime": 10,
        "spawnerSize": "APC",
        "type": "ASLAV",
        "vehTags": [
            "Class_Heavy", // new
            "Watercraft"   // new
        ],
        "vehType": "IFV"
    },
```

## 3. Added Sea Level

Added SeaLevel extraction in `GrabAsset` (fix #122)

``` ...
    "minimapTexture": "T_AlBasrah_Minimap",
    "rawName": "AlBasrah_AAS_v2",
    "seaLevel": 0,
    ...
```

<img width="210" height="90" alt="image" src="https://github.com/user-attachments/assets/f62e52aa-25e9-48f1-b4f5-d3db607cc3dd" />


## 4. Enhanced Logging

1. Added logs to follow progress of the layer extraction in `GrabAsset`.
2. Fixed typos ("mpas" -> "maps")

Output in `/grabAssets/error.txt` :

```
    [2025-8-30T20:30:31Z] Opening Layer 1/211 : /Game/Maps/Anvil/Gameplay_Layers/Anvil_Invasion_v1.Anvil_Invasion_v1
    [2025-8-30T20:30:39Z] Opening Layer 2/211 : /Game/Maps/Anvil/Gameplay_Layers/Anvil_RAAS_v1.Anvil_RAAS_v1
    [2025-8-30T20:30:46Z] Opening Layer 3/211 : /Game/Maps/Anvil/Gameplay_Layers/Anvil_RAAS_v2.Anvil_RAAS_v2
    [2025-8-30T20:30:53Z] Opening Layer 4/211 : /Game/Maps/Anvil/Gameplay_Layers/Anvil_Skirmish_v1.Anvil_Skirmish_v1
    [2025-8-30T20:31:0Z] Limited # of maps logged. Quitting game.
```

<img width="210" height="90" alt="image" src="https://github.com/user-attachments/assets/74ef56cd-8fe4-457b-be86-79816287989a" />


## 5. Disabled `Load Asset Blocking` 

In `Faction List logging`, disabled the 4 different calls to `Load Asset Blocking` because it was breaking the default units of each factions on every layer.I'm not good enough to find the error so i disabled it waiting to find a solution

<img width="200" height="100" alt="image" src="https://github.com/user-attachments/assets/1a194ff8-9172-48eb-9622-8b63f7ee653c" />


Without the ""correction"" :

```
    "team1Units": [
        {
          "factionID": "ADF",
          "defaultUnit": "None",
          "types": [
            {
              "unitType": "Mechanized",
              "unit": "ADF_LO_Mechanized"
            },
        }
    ]
```

## 6. Added more invalid map filters

The last UE5 SDK update bring a lot of weird "Co-op" layers that are not playable and seems to be for testing purpose only.
I added two filters on "Test" & "Coop" that remove 30 unwanted layer from the script, but that won't catch some weird layers like "Narva_SearchAndDestroy.Narva_SearchAndDestroy" or "Mutaha_Frontline.Mutaha_Frontline"

<img width="200" height="150" alt="image" src="https://github.com/user-attachments/assets/109816bc-73ba-47d9-87a4-93730f3964b4" />

sh4rkman committed:
Update MapGrabAsset4_0.uasset

- Sea Level (#122)
- Vehicle Tags
- boxExtent scaling (#121)
- Enhanced Logging

sh4rkman committed:
removed limiter

sh4rkman committed:
AAS links

to cover for the new skirmish layer now exporting links from all AAS Graphs layer

Shanomac99 commented:
Overall looks really good,

However, I do not like the addition of number 2. I think adding this would add too much bloat and would better fit for https://github.com/Squad-Wiki/squad-wiki-pipeline-weapon-and-vehicle-data/tree/main when vehicles are added. I'll remove 2 here, and then add #123 to a separate push.

Shanomac99 committed:
Change Invalid Layer Detection

Change invalid layer detection to run off an array instead of multiple checks
Changed invalid layer detection to look at entire path and not just display name (helps with removing Coop Maps

Shanomac99 committed:
Fixed issue with offset/empty units

Order was incorrect for asset loading, nodes tried to pull name and then load asset which is incorrect. This caused the node to use the previous loop iterations data or display None.

Reordered nodes so that the load asset blocking happens before the name is grabbed and saved.

Shanomac99 committed:
Implement singleUse logging of vehicles

Fixes #124
Implements singleUse within the vehicle data of the units, if true then the vehicle only spawns once up to the count.

Shanomac99 committed:
Corrected Layer Count

Added one to the layer count as its "loading" into the next map which isn't counted in the current maps JSON.

Also set limiter to 9999.

Shanomac99 committed:
Data Run for 9.0

sh4rkman commented:
Maybe Bump the .uasset name from 4.0 to 5.5 ?
I'm getting lost in all the versions

Shanomac99 commented:
4.0 is the version of the extractor, it's the 4 iteration. We could bump to a 4.1.

Shanomac99 committed:
Add initial change log.
Calling this update 2.1.

Shanomac99 committed:
Changed skirmish to show correct data.

- Version bump of MapGrabAssets to 4.1
- Added fallback for invalid gamemode
- Changed the way skirmish is logged in `capturePoints` (closes #126)
    - Changed `type` from `"AAS Graph"` to `"Skirmish Graph"`
    - Changed pointsOrder to have `invalidForSkirmishGameMode` listed when in skirmish as gamemode can have multiple points connected to one.
    - Fixed numberOfPoints calculation

sh4rkman committed:
Map Settings & Player Spawns

 - Added `SeaLevel` (closes #122)
  - Added `commanderDisabled`, if false commander is disabled on the layer
  - Added `boatsAvailable` boolean (informative only)
  - Added `tanksAvailable` boolean (informative only)
  - Added `helicoptersAvailable` boolean (informative only)
  - Added Player Spawn Objects within `mapAssets[]` (close #90)
- updated wiki screenshot for UE5

sh4rkman commented:
Last commit (381c39f ) now get helicopers/tanks/boats/commander availability (i found out after i added it that it was already here but the code wasn't being used, maybe there's an history i dont have ?). 

Looking back at it, i might have create a .settings[] for them in the map Object.

<img width="150" height="200" alt="image" src="https://github.com/user-attachments/assets/f99af565-37b8-4b7b-a68e-c430b7246f37" />

---

mapAssets now holds `spawnGroups` & `spawnPoints` in **separate** arrays.

<img width="200" height="320" alt="image" src="https://github.com/user-attachments/assets/b59fbf50-0889-48cf-ba9e-cbb722e2346a" />

sh4rkman committed:
updated export file

Shanomac99 committed:
Update finished.json to 9.0.2
